odoo/addons/base/models/ir_model.py

```diff
@@ -156,6 +160,24 @@ def _check_model_name(self):
             if not models.check_object_name(model.model):
                 raise ValidationError(_("The model name can only contain lowercase characters, digits, underscores and dots."))
 
+    @api.constrains('order', 'field_id')
+    def _check_order(self):
+        for model in self:
+            try:
+                model._check_qorder(model.order)  # regex check for the whole clause ('is it valid sql?')
+            except UserError as e:
+                raise ValidationError(e)
+            stored_fields = model.field_id.filtered('store').mapped('name')
+            if self.env.get(model.model) is None:
+                # model hasn't been init'd yet, which means that some fields are not yet in its
+                # list of fields but will be right after its creation - these fields can be used
+                # for ordering, so let's add them to the list of stored fields manually
+                stored_fields += models.MAGIC_COLUMNS
+            order_fields = re.findall(r'"?(\w+)"?\s*(?:asc|desc)?', model.order, flags=re.I)
```

Comment:
Tip: pre-compile the regexp at the module level
```python
RE_ORDER_FIELDS = re.compile(r'"?(\w+)"?\s*(?:asc|desc)?', re.I)
...
order_fields = RE_ORDER_FIELDS.findall(model.order)
```

Reply:
i feel like we're optimizing the operation that takes .0001% of this process but that's a good reflex anyway indeed